stan/math/opencl/prim/mdivide_left_tri_low.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef STAN_MATH_OPENCL_PRIM_MDIVIDE_LEFT_TRI_LOW_HPP
+#define STAN_MATH_OPENCL_PRIM_MDIVIDE_LEFT_TRI_LOW_HPP
+
+#include <stan/math/prim/mat/err/check_square.hpp>
+#include <stan/math/prim/mat/err/check_multiplicable.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/multiply.hpp>
+#include <stan/math/opencl/tri_inverse.hpp>
+namespace stan {
+namespace math {
+
+/**
+ * Returns the solution of the system Ax=b when A is lower triangular.
+ * @tparam T1 type of elements in A
+ * @tparam T2 type of elements in b
+ * @param A Triangular matrix.
+ * @param b Right hand side matrix or vector.
+ * @return x = A^-1 b, solution of the linear system.
+ * @throws std::domain_error if A is not square or the rows of b don't
+ * match the size of A.
+ */
+template <typename T1, typename T2,
+          typename = enable_if_all_floating_point<T1, T2>>
+inline matrix_cl<return_type_t<T1, T2>> mdivide_left_tri_low(
+    const matrix_cl<T1>& A, const matrix_cl<T2>& b) {
+  check_square("mdivide_left_tri_low", "A", A);
+  check_multiplicable("mdivide_left_tri_low", "A", A, "b", b);
+  return tri_inverse<matrix_cl_view::Lower>(A) * b;
```

Comment:
In order to force the Lower I had to extend the tri_inverse with a template.

The other option would be to force  a change of A before calling tri_inverse but that changes the view for A globally, which is bad. 

The third option would be to have the "forced" view as an argument to tri_inverse. I am also fine with that.

Reply:
I like the option you chose.